### PR TITLE
Re-work input parsing and validation

### LINKED ISSUE #6: Add support for more argument types
New argument types need to be supported to provide more flexibility to plugin developers. The following argument types should be supported. 

- uint16
- uint32
- int16
- int32
- int64
- float32
- float64

### PR DESCRIPTION
**Pull request by echlebek**
This CL uses spf13/viper for parsing configuration, rather than using a hand-rolled implementation. For parsing annotations, JSON is used.

Closes #6 
Closes #8 

Signed-off-by: Eric Chlebek <eric@sensu.io>

### CODE REVIEW

File: sensu/goplugin.go [hunk 1]
```diff
@@ -22,14 +24,32 @@ type GoPlugin interface {
 // PluginConfigOption defines an option to be read by the plugin on startup. An
 // option can be passed using a command line argument, an environment variable or
 // at for some plugin types using a configuration override from the Sensu event.
+//
+//
```
Comment (ccressent) [inline sensu/goplugin.go]: Why the 2 extra lines?

File: sensu/goplugin.go [hunk 2]
```diff
@@ -22,14 +24,32 @@ type GoPlugin interface {
...
+	Argument string
+
+	// Shorthand is the shorthand command line argument to consult when parsing config.
+	Shorthand string
+
+	// Default is the default value of the config option.
+	Default interface{}
+
+	// Usage adds help context to the command-line flag.
+	Usage string
```
Comment (ccressent) [inline sensu/goplugin.go]: Thank you for the doc strings!

File: sensu/goplugin.go [hunk 3]
```diff
@@ -82,89 +102,131 @@ func (goPlugin *basePlugin) readSensuEvent() error {
...
+		if !ok {
+			return fmt.Errorf("only pointer to map[string]string is allowed, not %v", kind)
+		}
+		cmd.Flags().StringToStringVarP(ptr, opt.Argument, opt.Shorthand, viper.GetStringMapString(opt.Argument), opt.Usage)
+	case reflect.Slice:
+		ptr, ok := opt.Value.(*[]string)
+		if !ok {
+			return fmt.Errorf("only pointer to []string is allowed, not %v", kind)
+		}
+		cmd.Flags().StringSliceVarP(ptr, opt.Argument, opt.Shorthand, viper.GetStringSlice(opt.Argument), opt.Usage)
```
Comment (ccressent) [inline sensu/goplugin.go]: Nice!
We probably need to add a section on how to format the input to such types; eg: is it `--flag 1,2,3` for a slice flag?

File: sensu/goplugin_test.go [hunk 4]
```diff
@@ -61,6 +62,14 @@ func TestSetOptionValue_FalseBool(t *testing.T) {
...
 }
 
+func TestSetOptionValue_Int64(t *testing.T) {
+	var value int64
+	option := defaultOption1
+	option.Value = &value
+	assert.NoError(t, setOptionValue(&option, "42"))
+	assert.Equal(t, value, int64(42))
+}
+
```
Comment (ccressent) [inline sensu/goplugin_test.go]: Ultimately, maybe we should add test of every single one of these types, to make sure we don't inadvertently break any of them during a refactor?

File: .travis.yml [hunk 5]
```diff
@@ -1,10 +1,7 @@
 language: go
 go:
-- 1.12.x
-install:
-  - curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(go env GOPATH)/bin v1.17.1
+- 1.13.x
 script:
-  - golangci-lint run --disable-all --enable=vet --enable=ineffassign --enable=goconst --tests ./...
```
[the remark this answers is not in the document]
Comment (echlebek) [reply .travis.yml]: It failed to run in the allotted time provided, and vet is now run as part of testing, so I think we're better off without it.

File: sensu/goplugin.go (hunk 3 above)
Comment (echlebek) [reply sensu/goplugin.go]: Woof... probably

**Pull request merged**